packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -0,0 +1,1136 @@
+  /// widget.
+  ///
+  /// Will be called even if the interaction is disabled with
+  /// [disableTranslation], [disableScale], or [disableRotation].
+  ///
+  /// Wrapping the InteractiveViewer in a [GestureDetector] can also be used to
+  /// receive many gestures on the child, but [GestureDetector.onScaleStart],
+  /// [GestureDetector.onScaleUpdate], and [GestureDetector.onScaleEnd] will not
+  /// be called, so [onInteractionStart], [onInteractionUpdate], and [onInteractionEnd] should be used instead.
+  ///
+  /// The coordinates returned in the details are viewport coordinates relative
+  /// to the parent. See [TransformationController.toScene] for how to
+  /// convert the coordinates to scene coordinates relative to the child.
+  ///
+  /// See also:
+  ///  * [onInteractionStart]
+  ///  * [onInteractionEnd]
+  final GestureScaleUpdateCallback onInteractionUpdate;
+
+  /// A [TransformationController] for the transformation performed on the
+  /// child.
+  ///
+  /// Whenever the child is transformed, the [Matrix4] value is updated and all
+  /// listeners are notified. The value can also be set.
+  ///
+  /// {@tool dartpad --template=stateful_widget_material_ticker}
+  /// This example shows how transformationController can be used to animate the
+  /// transformation back to its starting position.
+  ///
+  /// ```dart
+  /// final TransformationController _transformationController = TransformationController();
+  /// Animation<Matrix4> _animationReset;
+  /// AnimationController _controllerReset;
+  ///
+  /// void _onAnimateReset() {
+  ///   setState(() {
+  ///     _transformationController.value = _animationReset.value;
+  ///   });
+  ///   if (!_controllerReset.isAnimating) {
+  ///     _animationReset?.removeListener(_onAnimateReset);
+  ///     _animationReset = null;
+  ///     _controllerReset.reset();
+  ///   }
+  /// }
+  ///
+  /// void _animateResetInitialize() {
+  ///   _controllerReset.reset();
+  ///   _animationReset = Matrix4Tween(
+  ///     begin: _transformationController.value,
+  ///     end: Matrix4.identity(),
+  ///   ).animate(_controllerReset);
+  ///   _controllerReset.duration = const Duration(milliseconds: 400);
+  ///   _animationReset.addListener(_onAnimateReset);
+  ///   _controllerReset.forward();
+  /// }
+  ///
+  /// // Stop a running reset to home transform animation.
+  /// void _animateResetStop() {
+  ///   _controllerReset.stop();
+  ///   _animationReset?.removeListener(_onAnimateReset);
+  ///   _animationReset = null;
+  ///   _controllerReset.reset();
+  /// }
+  ///
+  /// void _onInteractionStart(ScaleStartDetails details) {
+  ///   // If the user tries to cause a transformation while the reset animation is
+  ///   // running, cancel the reset animation.
+  ///   if (_controllerReset.status == AnimationStatus.forward) {
+  ///     _animateResetStop();
+  ///   }
+  /// }
+  ///
+  /// IconButton get resetButton {
```

Comment:
I think it would be best to not suggest factoring Widgets with functions. It's easy enough to just create a const widget instead:

```dart
/// class _ResetButton extends StatelessWidget {
///   const _ResetButton({ Key key }) : super(key: key);
///   @override
///   Widget build(BuildContext context) {
///     return IconButton(
///       onPressed: () {
///         setState(() {
///           _animateResetInitialize();
///         });
///       },
///       tooltip: 'Reset',
///       color: Theme.of(context).colorScheme.surface,
///       icon: const Icon(Icons.replay),
///     );
///   }
/// );
```